File: server/src/elsewise/settings/languages.py
from typing import Literal

type SupportedLanguage = Literal["en", "ru", "fr", "es", "de", "pt-BR"]
SUPPORTED_LANGUAGES: tuple[SupportedLanguage, ...] = (
    "en",
    "ru",
    "fr",
    "es",
    "de",
    "pt-BR",
)
SUPPORTED_LANGUAGE_SET = frozenset(SUPPORTED_LANGUAGES)
LANGUAGE_DISPLAY_NAMES: dict[SupportedLanguage, str] = {
    "en": "English",
    "ru": "Русский",
    "fr": "Français",
    "es": "Español",
    "de": "Deutsch",
    "pt-BR": "Português (Brasil)",
}

File: web/src/i18n/languages.ts
export const SUPPORTED_LANGUAGES = [
  "en",
  "ru",
  "fr",
  "es",
  "de",
  "pt-BR",
] as const;

export type SupportedLanguage = (typeof SUPPORTED_LANGUAGES)[number];

export const LANGUAGE_LABELS: Record<SupportedLanguage, string> = {
  en: "English",
  ru: "Русский",
  fr: "Français",
  es: "Español",
  de: "Deutsch",
  "pt-BR": "Português (Brasil)",
};

export function isSupportedLanguage(value: string): value is SupportedLanguage {
  return SUPPORTED_LANGUAGES.includes(value as SupportedLanguage);
}
